### PR TITLE
statisitcs: estimate range result with increase factor.

### PR DESCRIPTION
**Pull request by hanfei1991**
If we analyze in sometime and insert a lot of value after that, the table count may be far larger than the count in histogram. So we must estimate by multiplying a increase factor.

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: statistics/histogram.go [hunk 1]
```diff
@@ -323,6 +323,15 @@ func (hg *Histogram) mergeBuckets(bucketIdx int64) {
 	return
 }
 
+func (hg *Histogram) getIncreaseFactor(totalCount int64) float64 {
```
Comment (shenli) [inline statistics/histogram.go]: Add comment for this function.

File: statistics/table.go [hunk 2]
```diff
@@ -155,45 +155,57 @@ func (t *Table) String() string {
...
-func (t *Table) columnIsInvalid(colInfo *model.ColumnInfo) bool {
+// ColumnIsInValid checks if this column is invalid.
+func (t *Table) ColumnIsInValid(colInfo *model.ColumnInfo) bool {
 	if t.Pseudo {
 		return true
 	}
-	_, ok := t.Columns[colInfo.ID]
-	return !ok
+	col, ok := t.Columns[colInfo.ID]
+	return !ok || len(col.Buckets) == 0
```
Comment (shenli) [inline statistics/table.go]: If there is nothing in this column or table, is ColumnIsInValid == true?

Comment (hanfei1991) [inline statistics/table.go]: yes

**Comment by hanfei1991** [comment]
@shenli @zimulala PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**